spyder/utils/syntaxhighlighters.py

```diff
@@ -422,6 +424,11 @@ def __init__(self, parent, font=None, color_scheme='Spyder'):
         self.import_statements = {}
         self.found_cell_separators = False
         self.cell_separators = CELL_LANGUAGES['Python']
+        # Avoid updating the outline explorer with every single letter typed
+        self.outline_explorer_data_update_timer = QTimer()
+        self.outline_explorer_data_update_timer.setSingleShot(True)
+        self.outline_explorer_data_update_timer.timeout.connect(
+                self.sig_outline_explorer_data_changed)

```

Comment:
4 spaces for hanging indents (Spyder is wrong in that regard, I *still* need to fix it...)